core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/NumericEditorDirective.js

```diff
@@ -0,0 +1,127 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for a numeric input question in the simple editor.
+ */
+
+// NOTE TO DEVELOPERS: This is meant to be a reusable directive, so its only
+// dependencies should be standard utility services. It should not have any
+// concept of "state in an exploration".
+oppia.directive('numericEditor', [
+    'QuestionIdService', 'AnswerGroupObjectFactory', 'RuleObjectFactory',
+    'StatusObjectFactory', 'OutcomeObjectFactory', 'UrlInterpolationService',
+    function(QuestionIdService, AnswerGroupObjectFactory, RuleObjectFactory,
+      StatusObjectFactory, OutcomeObjectFactory,UrlInterpolationService) {
+      return {
+        restrict: 'E',
+        scope: {
+          // A unique ID that allows events to be broadcast specifically to this
+          // directive.
+          getUniqueId: '&uniqueId',
+          getCustomizationArgs: '&customizationArgs',
+          getAnswerGroups: '&answerGroups',
+          getRawDefaultOutcome: '&defaultOutcome',
+          saveCustomizationArgs: '&',
+          saveAnswerGroups: '&',
+          saveDefaultOutcome: '&',
+          addState: '&'
+        },
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/exploration_editor/simple_editor_tab/questions/' +
+          'numeric_editor_directive.html'),
+        controller: [
+          '$scope', '$timeout', 'alertsService',
+          function($scope, $timeout, alertsService) {
+            // Note that a questionId generated in this way may contain spaces,
+            // since it is just the state name.
+            $scope.questionId = $scope.getUniqueId();
+            var answerGroups = $scope.getAnswerGroups();
+
+            $scope.getSubfieldId = function(label) {
+              return QuestionIdService.getSubfieldId($scope.questionId, label);
+            };
+
+            $scope.getFieldId = function(index) {
+              return $scope.questionId + '.' + index;
+            };
+
+            $scope.getAnswer = function() {
+              var newAnswerGroups = $scope.getAnswerGroups();
+              if (answerGroups.length !== 0) {
+                return newAnswerGroups[0].rules[0].inputs.x;
+              }
+            };
+
+            $scope.saveAnswer = function(newAnswer) {
+              var newAnswerGroups = answerGroups;
+              var validValue = /^[0-9]+$/;
```

Comment:
Decimals? Negative numbers? Can we also instead use the existing filter that's currently used for the learner view in the numeric input field, so that we have a consistent definition of validity throughout the application?